contracts/VeListaAutoCompounder.sol

```diff
@@ -0,0 +1,287 @@
+    function initialize(
+        address _lista,
+        address _velista,
+        address _veListaDistributor,
+        address _oracle,
+        address _feeReceiver,
+        address _admin,
+        address _bot
+    ) public initializer {
+        require(
+            _lista != address(0) &&
+                _velista != address(0) &&
+                _veListaDistributor != address(0) &&
+                _feeReceiver != address(0) &&
+                _admin != address(0) &&
+                _bot != address(0) &&
+                _oracle != address(0),
+            "Zero address provided"
+        );
+        __AccessControl_init();
+        _setupRole(DEFAULT_ADMIN_ROLE, _admin);
+        _setupRole(BOT, _bot);
+
+        lista = IERC20(_lista);
+        veLista = IVeLista(_velista);
+        require(lista == veLista.token(), "Invalid Lista token");
+
+        veListaDistributor = IVeListaDistributor(_veListaDistributor);
+        require(veLista == veListaDistributor.veLista(), "Invalid veLista");
+
+        oracle = OracleInterface(_oracle);
+        feeReceiver = _feeReceiver;
+
+        feeRate = 300; // 3%
+        maxFeeRate = 1000; // 10%
+        require(
+            feeRate <= maxFeeRate && maxFeeRate <= 10000,
+            "Invalid fee rate"
+        );
+
+        autoCompoundThreshold = 5 * 1e18; // $5
+        maxFee = 10 * 1e18; // $10
+        require(
+            (feeRate * autoCompoundThreshold) / 10000 <= maxFee,
+            "Invalid fee"
+        );
+
+        emit VeListaDistributorUpdated(_veListaDistributor);
+        emit OracleUpdated(_oracle);
+        emit FeeReceiverUpdated(_feeReceiver);
+    }
+
+    /**
+     * @dev Auto-compound the veLista rewards for an account; only callable by Bot
+     *      Claim LISTA from veListaDistributor and then compound LISTA by doing `increaseAmountFor`.
+     *
+     */
+    function claimAndIncreaseAmount(address account, uint16 toWeek) public onlyRole(BOT) {
+        require(autoCompoundEnabled[account], "Auto compound not enabled");
+
+        IVeLista.AccountData memory data = veLista.getLockedData(account);
+        require(data.locked > 0, "No locked amount");
+
+        (uint256 claimableAmt, ) = veListaDistributor.getTokenClaimable(
+            account,
+            address(lista),
+            toWeek
+        );
+        require(
+            claimableAmt > 0 && isEligibleForAutoCompound(claimableAmt),
+            "Not eligible for auto compound"
+        );
+
+        uint256 claimedAmount = veListaDistributor.claimForCompound(
+            account,
+            address(lista),
+            toWeek
+        );
+        uint256 amtToCompound = getAmountToCompound(claimedAmount);
+
+        IVeLista(veLista).increaseAmountFor(account, amtToCompound);
+
+        totalFee += claimedAmount - amtToCompound;
+
+        emit AutoCompounded(account, address(lista), claimedAmount, amtToCompound);
+    }
+
+    function withdrawFee() external onlyRole(DEFAULT_ADMIN_ROLE) {
```

Comment:
change the role to bot

Reply:
done in 0f24cf5aea994a3e2e7782ea799ba1b1d6309c6a

Reply:
changed the permission to Bot in 60c7c0374e641c87c77eed5a93f18acdf67d280c